src/business-logic/services/superfly-timeline-builder.ts

```diff
@@ -63,7 +68,7 @@ export class SuperflyTimelineBuilder implements TimelineBuilder {
     timeline = this.createTimelineWithInfiniteGroups(rundown, timeline)
     timeline = this.updateTimelineAutoNextEpochTimestampFromNextPart(rundown, activePartTimelineGroup, timeline)
 
-    return timeline
+    return Promise.resolve(timeline)
```

Comment:
Either the `Promise.resolve` or `async` should go from this method.

Reply:
Removing the async makes Jest unable to catch errors in test and assert them.
Removing the Promise.resolve makes lint complaint about us not awaiting anyting.
Removing Promise from the return type is not an option.

I am up for suggestions

Reply:
Beats me

Reply:
Removing the async keyword makes it such that thrown errors are not picked up and converted to a rejected promise.
In order to remove async, the method body should be wrapped in a try-catch that converts it to a rejected promise.
I'll let it be up to you whether to keep async or do the promise conversion.

Reply:
Keeping the async then